VetClinic.API.Tests/Controllers/PositionControllerTests.cs

```diff
@@ -0,0 +1,197 @@
+﻿using AutoFixture;
+using AutoFixture.Xunit2;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using VetClinic.API.Controllers;
+using VetClinic.API.DTO.PositionDTO;
+using VetClinic.BLL.Services.Interfaces;
+using VetClinic.DAL.Entities;
+using Xunit;
+
+namespace VetClinic.API.Tests.Controllers
+{
+    public class PositionControllerTests
+    {
+        private readonly Mock<IPositionService> positionServiceMock;
+        private readonly Mock<IMapper> mapper;
+        private readonly PositionController positionController;
+        public PositionControllerTests()
+        {
+            var fixture = new Fixture();
+            positionServiceMock = fixture.Freeze<Mock<IPositionService>>();
+            mapper = fixture.Freeze<Mock<IMapper>>();
+            positionController = new PositionController(positionServiceMock.Object, mapper.Object);
+
+        }
+
+
+        [Theory, AutoMoqData]
+        public async Task GetAll_Positions_ReturnsTrue(
+            [Frozen] List<Position> positions,
+            [Frozen] List<PositionDto> positionsDTO)
+        {
+            // Arrange
+            positionServiceMock.Setup(m => m.GetPositionAsync())            
+                .ReturnsAsync(positions);
+            mapper.Setup(m => m.Map<ICollection<PositionDto>>(positions))
+                .Returns(positionsDTO);            
+
+            // Act
+            var actualResult = await positionController.Index();
+
+            // Assert      
+            var result = actualResult.Result as OkObjectResult; 
+            
+            Assert.Equal(positionsDTO, result.Value );
+            Assert.True(actualResult.Result is OkObjectResult);
+            positionServiceMock.Verify(m => m.GetPositionAsync(), Times.Once);
+        }
+
+
+        [Theory, AutoMoqData]
+        public async Task GetById_PositionId_ReturnsPositionWithRequestedId(
+           [Frozen] Position position,
+           [Frozen] PositionDto positionDTO )
+        {
+            // Arrange           
+            positionServiceMock.Setup(p => p.GetPositionAsync(position.Id))
+                .ReturnsAsync(position);
+            mapper.Setup(m => m.Map<PositionDto>(position))
+                .Returns(positionDTO);          
+
+            // Act
+            var actualResult = await positionController.Show(position.Id);
+
+            // Assert    
+            var result = actualResult.Result as OkObjectResult;
+            Assert.Equal(positionDTO, result.Value);
+            Assert.True(actualResult.Result is OkObjectResult);
+            positionServiceMock.Verify(m => m.GetPositionAsync(position.Id), Times.Once);
+        }
+
+
+        [Fact]
+        public async Task GetById_PositionId_ReturnsNotFound__withNewMock()
```

Comment:
Correct the name. Use names that are closer to the language of the business ( __withNewMock - what does it means)